File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> complement;
        vector<int> result;
        for (int i = 0; i < nums.size(); ++i){
            int comp = target - nums[i];
            if (complement.find(comp) != complement.end()){
                result.push_back(complement[comp]);
                result.push_back(i);
                break;
            }
            complement[nums[i]] = i;
        }
        return result;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        diffs = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if nums[i] in diffs:
                return [i, diffs[nums[i]]]
            diffs[diff] = i

        